lib/jekyll/site.rb

```diff
@@ -363,6 +363,7 @@ def each_site_file
       pages.each { |page| yield page }
       static_files.each { |file| yield(file) if file.write? }
       collections.each_value { |coll| coll.docs.each { |doc| yield(doc) if doc.write? } }
```

Comment:
This would have a lower cyclomatic complexity, what do you think? 
```ruby
 collections.each_value do  |coll| 
   if doc.write?
     coll.docs.each(&method(:yield))
     coll.files.each(&method(:yield)) 
  end
end
```